ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Keystore/Types.hs

```diff
@@ -0,0 +1,95 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Ariadne.Wallet.Cardano.Kernel.Keystore.Types
+       ( StorageDecodingError (..)
+       , InternalStorage (..)
+       , Version (..)
+       , isFilePath
+       , isWallets
+       , isVersion
+       , getInternalStorageKeys
+       ) where
+
+import Control.Lens.TH (makeLenses)
+import Data.Aeson
+  (FromJSON, ToJSON, Value(..), toJSON, parseJSON, withText, withObject, (.:), object, (.=))
+import Data.Aeson.Types (Parser)
+import Data.Default (def, Default)
+import Data.Version (Version(..), makeVersion, showVersion, parseVersion)
+import Data.Map (Map)
+import Formatting (bprint, build)
+import Text.ParserCombinators.ReadP (readP_to_S)
+
+import Pos.Binary.Class (serialize', decodeFull')
+import Pos.Core (AddressHash)
+import Pos.Crypto (EncryptedSecretKey(..), PublicKey)
+
+import  Ariadne.Wallet.Cardano.Kernel.Keystore.Util (esksToMap)
+
+import qualified Data.Text.Buildable
+import qualified Data.Map as Map (elems, empty)
+import qualified Serokell.Util.Base16 as S (encode, decode)
+
+-- | InternalStorage data containing wallet keys that gets saved to file
+data InternalStorage
+  = InternalStorage
+  { _isWallets  :: Map (AddressHash PublicKey) EncryptedSecretKey
+  , _isVersion  :: !Version
+  , _isFilePath :: !FilePath
+  } deriving (Show)
+
+makeLenses ''InternalStorage
+
+instance Default InternalStorage where
+  def = InternalStorage
+      { _isWallets  = Map.empty
+      , _isVersion  = makeVersion [0, 0, 0]
+      , _isFilePath = mempty
+      }
+
+newtype StorageDecodingError = StorageDecodingError Text
+    deriving (Show)
+
+instance Exception StorageDecodingError
+
+instance Buildable StorageDecodingError where
+    build (StorageDecodingError msg) =
+        "Failed to decode stored json: " <> bprint build msg
+
+instance ToJSON EncryptedSecretKey where
+  toJSON = String . S.encode . serialize'
+
+instance FromJSON EncryptedSecretKey where
+  parseJSON  = withText "EncryptedSecretKey" $ \encryptedText ->
+                let eitherDecode = either textFail pure . decodeFull'
+                in either textFail eitherDecode $ S.decode encryptedText
```

Comment:
Please don't indent code like this. Add a linebreak after `parseJSON =`, then add 4 spaces per indentation level. Also please make sure you use 4 spaces everywhere.